java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -46,30 +45,31 @@
 import org.sonar.plugins.java.api.tree.VariableTree;
 
 @Rule(key = "S1612")
-public class ReplaceLambdaByMethodRefCheck extends BaseTreeVisitor implements JavaFileScanner {
+public class ReplaceLambdaByMethodRefCheck extends SubscriptionVisitor {
```

Comment:
While I do appreciate the change, it has nothing to do with handling the FP in itself. 

For the sack of consistency and clarity when looking at changes, please split it in two commits:
1. Changing to a SubscriptionVisitor + all the style changes (missing spaces, changes of tabs), which are unrelated
2. Update the rule to handle the FP